teps/0057-windows-support.md

```diff
@@ -0,0 +1,154 @@
+
+Tekton Hub and Catalog should allow a Task to indicate that it requires a Windows node to run, if this is absent, the default behaviour will be to assume Linux so that no change to existing Linux workflows is required.
+
+## Proposal
+Extend Tekton so TaskRuns can request, and be scheduled, to run on Windows nodes. This will require (at least) reworking the following Tekton components:
+- Entrypoint binary
+- All internal support images used for resource management
+- File paths used for step ordering, either via polling or watching files
+- Script mode
+
+### Notes/Caveats (optional)
+In order to have Windows nodes run TaskRuns there are some elements of Tekton that will need reworking to be compatible with that OS. Some of these have been identified in the [issue](https://github.com/tektoncd/pipeline/issues/1826) but it is almost certain that other elements will need work in order to work. 
+
+### Risks and Mitigations
+- The burden of supporting the Windows platform may be quite high
+- Windows containers may introduce some security vulnerabilities
+
+### User Experience (optional)
+- User specifies via nodeSelector if they want to run a task in a Windows container
+    - It just works
+- The default (no nodeSelector specified) behaviour selects a Linux container
+    - No noticeable change for users who use Linux only TaskRuns.
+
+### Performance (optional)
+In the case where there are no Windows containers the goal is obviously no performance impact at all. When there are Windows containers then we would aim for equivalent performance as on Linux nodes, where possible.
+
+## Design Details
+TBD
+
+## Test Plan
+TBD
+
+## Design Evaluation
+### Reusability
+Windows support is definitely a feature that cannot be achieved using any existing features or by adding something to the catalog. However, the way in which we add Windows support cannot be limited to our unique use case. This nodeSelector functionality may provide a pathway for other platforms (e.g. if OSX containers ever become feasible).
+### Simplicity
+Windows support needs to work with everything else already in Tekton. We’d also need to examine our design and implementation plans carefully to see if there are edge cases which would add unnecessary complexity that we can avoid.
+### Flexibility
+As described above, Windows support would be explicitly opt-in, meaning that non-Windows users never need to know about this feature. Adding Windows support would actually improve the flexibility of Tekton significantly by supporting workflows that are dependent on Windows containers.
+### Conformance
+Other than having to specify that a given TaskRun needs to run on a Windows node there really shouldn’t be any significant user-facing changes. With that in mind, we intend that the user experience when creating TaskRuns for Windows nodes would be identical to the experience when creating Linux based TaskRuns, other than using nodeSelector to specify the need for a Windows node.
```

Comment:
i mentioned this in a comment above, but im wondering it makes more sense for this info to be in the Task vs the TaskRun (though the TaskRun might ultimately need to provide a nodeSelector as well 😬 )